### ISSUE TITLE
Wrong suggestions for "for" command

### ISSUE THREAD

**Issue by EmilySeville7cfg** [akinomyoga/ble.sh#156]
**ble version**: 0.4.0-devel3+82f74f0
**Bash version**: 5.0.17(1)-release (x86_64-pc-linux-gnu)
**Terminal version**: GNOME Terminal 3.36.2
**OS version**: Linux Mint 20.2 Cinnamon

There are no flags `-in`|`-do` but they are suggested:

![image](https://user-images.githubusercontent.com/42812113/144424162-db8730c8-ca57-4b8d-8684-57a82cb9694e.png)

**Comment by akinomyoga** [comment]
It doesn't appear as the completions for options. `-` after the keyword `for` is invalid, so the normal completion cannot be started. Instead, a completion for a new word is attempted assuming that `-` is a broken variable name. `in` and `do` are one of the possible second arguments in that case. See the following valid examples:

```bash
for a in {1..10}; do test; done
for a do echo $a; done
```

Nevertheless, maybe it is more sensible to automatically introduce a space after `-`, or just do not produce any completions.

**Comment by EmilySeville7cfg** [comment]
I think it's better add space not to confuse users. ;)

**Comment by akinomyoga** [comment]
Probably I am going to just turn off the completion in this context.

> The another option also print error message under current line like elvish shell does. In this case user will see instant feedback.

Hmm, may I confirm whether you see the highlighting for the syntax error just as in the below image? If you input a hyphen after the keyword `for`, that hyphen (or a later character if there is any succeeding characters)? At that moment, the user is supposed to notice that the hyphen cannot be there.

![image](https://user-images.githubusercontent.com/8982192/144427065-cf1708f1-11d4-45a0-bbf0-bc5f38b84cd1.png)

Though, I admit that the highlighting for the syntax error is overridden by the highlighting of the temporarily inserted completions from the menu.

For the description of the details of the error, I have a plan to implement them in the future (as mentioned in [the roadmap section of README](https://github.com/akinomyoga/ble.sh#history-and-roadmap)), but I think it will be far future.

**Comment by akinomyoga** [comment]
> Probably I am going to just turn off the completion in this context.

In commit 371a5a4, I have removed the completion settings of new words immediately after invalid words. Now `for -[TAB]` will not produce any completions.

> Hmm, may I confirm whether you see the highlighting for the syntax error

For this issue, I actually found that the check for invalid variable name for `for` statements have actually not been implemented. Also, somehow the highlighting based on filename, etc. is turned on. In commit d056547, I have made the check of the variable name a bit more strict and also turned off the highlighting based on filenames, etc.
